modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java

```diff
@@ -207,18 +207,12 @@ public String getDeleteFolderURL() throws PortalException {
 		return portletURL.toString();
 	}
 
-	public String getDownloadFolderURL() {
-		LiferayPortletResponse liferayPortletResponse =
-			_dlRequestHelper.getLiferayPortletResponse();
-
-		ResourceURL resourceURL = liferayPortletResponse.createResourceURL();
-
-		resourceURL.setParameter("folderId", String.valueOf(_getFolderId()));
-		resourceURL.setParameter(
-			"repositoryId", String.valueOf(_getRepositoryId()));
-		resourceURL.setResourceID("/document_library/download_folder");
+	public String getDownloadFolderSevenZipURL() {
+		return _getDownloadFolderURL("7z");
```

Comment:
_.7z_ as a parameter would be better?

Reply:
I still think "7z" is better. Please explain why ".7z" is better for you?